tools/src/launcher/Main.hs

```diff
@@ -202,6 +208,26 @@ main = do
                     , loUpdateArchive)
                     loNodeTimeoutSec
                     loReportServer
+  where
+    addConfigurationOptions :: ConfigurationOptions -> [Text] -> [Text]
+    addConfigurationOptions ConfigurationOptions {..} =
+        addConfFileOption cfoFilePath .
+        addConfKeyOption cfoKey . addSystemStartOption cfoSystemStart
+
+    addConfFileOption filePath =
+        maybeAddOption "--configuration-file" (toText filePath)
+    addConfKeyOption key = maybeAddOption "--configuration-key" key
+    addSystemStartOption =
+        maybe identity (maybeAddOption "--system-start" . timestampToText)
+
+    maybeAddOption :: Text -> Text -> [Text] -> [Text]
```

Comment:
won't it fail if somebody writes `--configuration-key=blah` instead of `--configuration-key blah`?

Reply:
Well, it will fail, but actually this logic wasn't even planned initially. I wanted to simply propagate all configuration options, but then decided to also check whether they are passed using `-n` argument which is slightly better, but doesn't cover all cases. I think it's ok here.

Reply:
Okay, yes, it seems ok.